src/main/java/org/cloudfoundry/promregator/endpoint/MetricsEndpoint.java

```diff
@@ -63,6 +67,10 @@ protected boolean isLabelEnrichmentSuppressable() {
 	 */
 	@Override
 	protected List<MetricsFetcher> createMetricsFetchers(List<Instance> instanceList) {
+		if(maxAllowedForSingleEndpoint != -1 && instanceList.size() > maxAllowedForSingleEndpoint) {
+			throw new RuntimeException("promregator.single.endpoint.max is configured to only allow "+maxAllowedForSingleEndpoint+" instances but there were "+ instanceList.size());
```

Comment:
Better use `String.format` here.

Reply:
Sorry I'm not sure I understand the reasoning. Is `String.format` preferred for performance reasons or for i18n?

I'm happy to oblige. I just want to understand why it is preferred.

Reply:
It's a kind of that 😄 

`String.format` is indeed supportive on i18n. However, Promregator is not i18n-enabled yet. I also don't expect to see happening that soon. Yet, I don't want to rule it out completely. If we then have many string concatenations all around, this wouldn't make our life easier. 

Moreover, I have seen too many cases (in my business life) where people started to "glue strings together", which they then spooled out as human readable text. By this they "hard-code the English grammar" (it's even worse when they do so writing "source code in German"). That is why I defined a rule on this to myself (which helped me getting along well so far): "If you have a placeholder in your text, then always use `String.format` (at least)". So, you might call it "experience out of pain" 😀 - or to nip things in the bud.

Is that reasoning okay with you?

Reply:
Certainly. The reason I asked is with slf4j and log4j2 you can do the following with logging: 

```
log.debug("Creating Metrics Fetcher for instance {}", instance.getInstanceId());
```

And I was considering making the recommendation eventually. (Obviously we would still stick to `String.format` for the exception messages though)

Reply:
Yeah, I know: for the logging facilities you have this alternative - which I would not disregard in general. However, to be honest, I am not used to this new feature - and I also had some bad experience (on serializing these kind of log records) in one project with it. That's why I am not using it so far (once bitten, twice shy).